pkg/cidata/cidata.TEMPLATE.d/boot/15-setup-dns.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+set -eux
+
+DNS=""
+# NOTE: Busybox sh does not support `for ((i=0;i<$N;i++))` form
+for f in $(seq 0 $((LIMA_CIDATA_DNS - 1))); do
+	dnsaddressvar="LIMA_CIDATA_DNS_${f}_ADDRESS"
+	dnsaddress="$(eval echo \$"$dnsaddressvar")"
+	if [ -z "${DNS}" ]; then
+		DNS="${dnsaddress}"
+	else
+		DNS="${DNS} ${dnsaddress}"
+	fi
+	echo "nameserver ${dnsaddress}" >>/etc/resolv.conf
```

Comment:
This should be only on Alpine, I guess

Reply:
and for other hosts, can we depend on cloud-init? Sed is not robust.

Reply:
> This should be only on Alpine, I guess

It is not needed on Alpine either. `udhcpc` will generate `/etc/resolv.conf` later via script, so any changes we make here are lost. I'm fine removing direct edits to `/etc/resolv.conf` completely, as I think they will likely never have any effect.

> can we depend on cloud-init

Absolutely! Trying to understand all the different ways the resolver can be configured seems to be a nightmare, which is why I asked if we need to support other mechanisms. Happy to delegate this to cloud-init (after checking that it actually works).